pkg/controller/dynamicupgradeableconditionscontroller.go

```diff
@@ -0,0 +1,245 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+	"github.com/go-logr/logr"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-olm-operator/pkg/clients"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	storage "github.com/operator-framework/helm-operator-plugins/pkg/storage"
+	"github.com/operator-framework/operator-registry/alpha/property"
+	helm "helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	incompatibleOperatorsInstalled = "IncompatibleOperatorsInstalled"
```

Comment:
Convention on reason constants is to prefix the identifier with `reason` (or `Reason` if we were exporting it)

```suggestion
	reasonIncompatibleOperatorsInstalled = "IncompatibleOperatorsInstalled"
```